app/services/return-requirements/check/persist-return-requirements.service.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+/**
+ * Persists the Return Requirements journey data
+ * @module PersistReturnRequirementsService
+ */
+
+const ReturnRequirementModel = require('../../../models/return-requirement.model.js')
+
+/**
+ * Persists the Requirements journey data based on session data
+ *
+ * Builds and persists a return version per return requirement
+ *
+ * Builds and persists the return requirements
+ *
+ * @param {module:SessionModel} session - The session for the return requirement journey
+ *
+ */
+async function go (session) {
+  const { startDateDay, startDateYear, startDateMonth, licence, reason, requirements } = session
+
+  const returnVersion = _buildReturnVersion(startDateDay, startDateYear, startDateMonth, licence, reason)
+  const returnRequirements = _buildReturnRequirements(requirements, returnVersion)
+
+  await _persistData(returnRequirements, returnVersion)
+}
+
+function _buildReturnVersion (startDateDay, startDateYear, startDateMonth, licence, reason) {
+  const startDate = startDateDay
+    ? new Date(`${startDateYear}-${startDateMonth}-${startDateDay}`)
+    : licence.currentVersionStartDate
+
+  return {
+    version: 1,
+    licenceId: licence.id,
+    startDate,
+    status: 'current',
+    reason
+  }
+}
+
+function _buildReturnRequirements (requirements, returnVersion) {
+  return requirements
+    // Strip empty objects off array - investigate bug
+    .filter((requirement) => { return Object.keys(requirement).length > 0 })
```

Comment:
Is this a bug at the start of the journey. 

We are seeing an empty object at the start of the array `[{}, { "this would have the journey data" }]`